tests/unit/core/test_processor.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+import os
+import pyarrow.parquet as pq
+from fastapi import HTTPException
+from tests.resources import test_data
+from data_service.config import config
+from data_service.core.processor import Processor
+
+
+RESULTSET_DIR = 'tests/resources/resultset'
+
+processor = Processor(
+    settings=config.LocalFileSettings(
+        DATA_SERVICE_URL='https://fake-data-service-url',
+        DATASTORE_DIR='tests/resources/datastore_unit_test',
+        RESULTSET_DIR=RESULTSET_DIR
+    )
+)
+
+
+def test_valid_event_request():
+    file_name = processor.process_event_request(
+        test_data.VALID_EVENT_QUERY_PERSON_INCOME_ALL
+    )
+    assert result_set_to_csv_string(file_name) == test_data.PERSON_INCOME_ALL
+
+
+def test_invalid_event_request():
+    with pytest.raises(HTTPException) as e:
+        processor.process_event_request(
+            test_data.INVALID_EVENT_QUERY_INVALID_STOP_DATE
+        )
+    assert "Empty result set" in e.value.detail
+    assert e.value.status_code == 204
```

Comment:
Is this logic visible in OpenAPI docs?

Reply:
It does not :/ Will try adding some app level exception handling that will autogenerate docs 🏃🏻